proto-lens-benchmarks/src/Data/ProtoLens/BenchmarkUtil.hs

```diff
@@ -0,0 +1,77 @@
+-- Copyright 2016 Google Inc. All Rights Reserved.
+--
+-- Use of this source code is governed by a BSD-style
+-- license that can be found in the LICENSE file or at
+-- https://developers.google.com/open-source/licenses/bsd
+
+{-# LANGUAGE ScopedTypeVariables #-}
+
+-- | A utility library for writing proto-lens benchmarks.
+module Data.ProtoLens.BenchmarkUtil (protoBenchmark, benchmarkMain) where
+
+import Control.DeepSeq (NFData)
+import Criterion.Main
+import qualified Criterion.Main.Options as Criterion
+import qualified Data.ByteString as BS (length)
+import Data.Maybe (fromMaybe)
+import Data.ProtoLens
+import Options.Applicative
+
+-- | Generate a group of benchmarks for encoding and decoding the given proto
+-- message. Includes benchmarks for decoding to both weak head normal form and
+-- normal form (i.e. all fields fully forced.) There is only one encoding
+-- benchmark because protos are encoded as strict ByteStrings, so there is no
+-- difference between normal form and weak head normal form.
+protoBenchmark
+    :: forall a.
+       (NFData a, Message a)
+    => String
+    -- ^ Benchmark name.
+    -> a
+    -- ^ Protocol buffer message to encode/decode.
+    -> Benchmark
+protoBenchmark groupName proto =
+    bgroup
+        fullGroupName
+        [ bench "encode" $ nf encodeMessage proto
```

Comment:
You mean just to avoid the boilerplate instances for Generic and NFData? Unfortunately I don't think it adds much value, because I think we want the encoding and decoding benchmarked separately so if there's a big improvement/regression we know which side it came from. And if we have separate encoding/decoding benchmarks, there's not much point in having the combined round-trip version.

It wouldn't be much work to add them though, so if you can think of a good reason to include them I'd be happy to add them.